src/ekf.cpp

```diff
@@ -362,7 +362,26 @@ namespace RobotLocalization
     estimateErrorCovariance_ = (transferFunctionJacobian_ *
                                 estimateErrorCovariance_ *
                                 transferFunctionJacobian_.transpose());
-    estimateErrorCovariance_.noalias() += (processNoiseCovariance_ * delta);
+
+    // Compute processNoiseCovariance scaling if nessesary
+    Eigen::MatrixXd processNoiseCovariance = processNoiseCovariance_;
+    if (enableProcessNoiseCovarianceScaler_)
+    {
+      // Scale covariance in the direction of the velocity vector
+      Eigen::MatrixXd velScaler = state_.segment<6>(StateMemberVx).asDiagonal() * processNoiseCovarianceScaler_;
+
+      Eigen::Matrix3d R = (Eigen::AngleAxisd(state_(3), Eigen::Vector3d::UnitX())
```

Comment:
I guess you can replace 3, 4, 5 with the enums for the corresponding state names.